src/commands.js

```diff
@@ -1,59 +1,29 @@
 const sleep = require('sleep');
-
+const {TextChannel, GuildChanne} = require('discord.js')
 //Clears the chat logs
-function clearMessages(message)
-{
-  //Gets all messages in the bot-commands channel
-  message.channel.fetchMessages().then(function(messages)
-  {
-    //Bulk deletes all the messages
-    message.channel.bulkDelete(messages);
-  });
-
-  //Sends a message
-  message.channel.send("Chat log has been cleared!").then(function(thisMessage)
-  {
-    //After the message it sent, it waits three seconds and deletes itself
-    sleep.sleep(3);
-    thisMessage.delete();
-  });
+function bulkDelete(textchan) {
+	if (textchan instanceof TextChannel && textchan.guild.me.hasPermission("MANAGE_MESSAGES") || textchan.memberPermissions(textchan.guild.me).has("MANAGE_MESSAGES")) {
+		return true
+		//Gets all messages in the bot-commands channel
+		textchan.fetchMessages().then(function(messages) {
+		//Bulk deletes all the messages
+		textchan.bulkDelete(messages);
```

Comment:
Is this called recursively?

Reply:
So how exactly does it calling recursively work? Does it delete one at a time?

Reply:
Is it currently using the discord bulkDelete? If so, we should change the name of the function as to not confuse people.